modules/game/src/main/BSONHandlers.scala

```diff
@@ -107,7 +108,7 @@ object BSONHandlers:
               .filter(HalfMoveClock.initial <= _)
           PgnStorage.Decoded(
             sans = sans,
-            pieces = BinaryFormat.piece.read(r bytes F.binaryPieces, gameVariant),
+            board = BBoard.fromMap(BinaryFormat.piece.read(r bytes F.binaryPieces, gameVariant)),
```

Comment:
We can read/write with Bitboard directly here, but I guess this is only for legacy format, we shouldn't run into this often, so I haven't changed anything here.

But, it can be another PR if we need to optimize it.

Reply:
it's what handles all chess variants. rather negligible indeed